Create first-shopify-abtest.mdx (#172)
statsig-io/docs: docs/guides/first-shopify-abtest.mdx

---
sidebar_label: Your First Shopify A/B Test
title: Build your first Shopify A/B test  
---

> In this guide we'll run an A/B test to optimize the landing page hero banner of our Shopify store. Our hypothesis is that optimizing the hero banner will engage new visitors and returning buyers in a way that leads them to buy more from the store. We'll measure this using metrics that include 'add to cart' and purchase events from buyers. Let's dive in!

Reviewer: 'purchase' should be in quotes similar to 'add to cart'.

> In this guide we'll run an A/B test to optimize the landing page hero banner of our Shopify store. Our hypothesis is that optimizing the hero banner will engage new visitors and returning buyers in a way that leads them to buy more from the store. We'll measure this using metrics that include 'add to cart' and purchase events from buyers. Let's dive in!

Reviewer: ```suggestion
In this guide we'll run an A/B test to optimize the landing page hero banner of our Shopify store. Our hypothesis is that optimizing the hero banner will engage new visitors and returning buyers in a way that leads them to buy more from the store. We'll measure this using metrics that include 'add to cart' and 'purchase event' from buyers. Let's dive in!
```

I don't see a reference to add to cart, but there's a reference to purchase event below.

### Step 1: Set up your Shopify Store with the test variants
Here's a detailed guide to [creating your Shopify store](https://www.shopify.com/blog/start-online-store), adding the products you want to sell, and picking a theme. To follow along with this guide, you'll want to use a Dawn-based theme for your store. 

> Once your store is up and running, you'll want to log into your Shopify store as admin. You might already have done this and here's just a quick recap so we're all on the same page: 

Reviewer: ```suggestion
Once your store is up and running, log into your Shopify store as an admin. You might already have done this and here's just a quick recap so we're all on the same page: 
```

![image](https://user-images.githubusercontent.com/1315028/150182819-5a6e1745-7e54-4d02-a3e4-406eb278da8d.png)

- Repeat the previous two steps for the **Second Image** as well
- At this point, you should see both images in your hero banner on the store 

### Step 2: Create an experiment on Statsig

To create an experiment on Statsig, 
- Log into https://console.statsig.com/
- Click on **Experiments** from the left hand navigation menu
- Click on **Create**, then enter the name and description of the experiment
- In the experiment **Setup** tab, enter the names of the variants that you want to test in the **Groups** panel; in this case I'm just calling them Control and Test. 
- Staying in the **Groups** panel, click on **sample_parameter** and enter a parameter name of your choice; in this case, I'm calling the parameter **image_selector**. Enter the parameter values that you want to use; in this case, I simply want to toggle between the two images that we've already uploaded to the store in the previous step

![image](https://user-images.githubusercontent.com/1315028/150181235-068a1cf3-00ef-48f8-b2fa-51c0338e5e6d.png)

> ### Step 3: Customize your theme

Reviewer 2: This is a super hacky way to do it, and run a specific test related to swapping images.  Worth noting that this doesnt just magically run any arbitrary experiments

After you log into your store in step 1, click on **Themes** under the **Online Store** in the left hand navigation panel. 
Then, 
- Click on the **Actions** drop down, then click on **Edit Code**
- Click on **theme.liquid** file under the **Layout** section
- Load the SDK by copying the code the below in the `head` section; use the experiment config to show the right image to your visitors

```js
<script src="https://cdn.jsdelivr.net/npm/statsig-js/build/statsig-prod-web-sdk.min.js"></script>
```

- Initialize the SDK and get the experiment config by copying the following in the `body` section

```js
<script>
    // hide the page until we fetch the configuration from statsig
    document.getElementsByTagName("BODY")[0].style.display = "none";
    
    //initialize the Statsig SDK using your Statsig client SDK key 
    window.statsig.initialize("<CLIENT_SDK_KEY>").then(() => {

>         // Assuming you're on the Dawn theme and you've uploaded two images for the hero image

Reviewer 2: people might glance over this but they need to put in an SDK key and an experiment name further down

        const heroImages = document.querySelectorAll(".banner > .banner__media");
        if (heroImages.length == 2) {
            heroImages[0].classList.remove("banner__media-half");
            heroImages[1].classList.remove("banner__media-half");
            
            //get experiment config, control (default) is 0
            const config = statsig.getExperiment("shopify_hero_image_abtest").get("image_selector", 0)

            // swap the images if the user is assigned to treatment  

>             if (config == 1){

Reviewer 2: 0 and 1 would be a strange way to do this, I would make this booleans

              const primaryHTML = heroImages[0].innerHTML;
              const primaryOuter = heroImages[0].outerHTML;
              
              heroImages[0].innerHTML = heroImages[1].innerHTML;
              heroImages[0].outerHTML = heroImages[1].outerHTML;
              
              heroImages[1].innerHTML = primaryHTML;
              heroImages[1].outerHTML = primaryOuter;
            }
        }
        // show the page
        document.getElementsByTagName("BODY")[0].style.display = "block";                                         
    });
</script>
```

- Log the key events that you want to track as metrics (along with any metadata) by copying the code below into the pages where customers take key actions

```js

> statsig.logEvent('purchase_event', 'SKU_12345', {'price': '9.99', 'item_name': 'diet_coke_48_pack'});

Reviewer 2: This is from other docs, but we should update the value to be the product category - thats our current suggestion, anyway
